#! /bin/sh /usr/share/dpatch/dpatch-run
## memcached_fdleak.dpatch by Sebastian Harl <sh@tokkee.org>
##
## DP: Fixed a possible file descriptor leak in the memcached plugin.
## DP: This has been reported as upstream bug #26:
## DP: http://collectd.org/mantis/view.php?id=26
## DP: (This is based on upstream Git commit
## DP: e7929dac268957cbbd9082717759c3917ac1b51e)

@DPATCH@

diff a/src/memcached.c b/src/memcached.c
--- a/src/memcached.c
+++ b/src/memcached.c
@@ -137,6 +137,8 @@ static int memcached_query_daemon (char *buffer, int buffer_size) /* {{{ */
 
 		if (n <= 0) {
 			ERROR ("memcached: poll() failed or timed out");
+			shutdown (fd, SHUT_RDWR);
+			close (fd);
 			return -1;
 		}
 	}
